web/packages/teleterm/src/ui/services/connectionTracker/connectionTrackerService.ts

```diff
@@ -162,6 +163,27 @@ export class ConnectionTrackerService extends ImmutableStore<ConnectionTrackerSt
     });
   }
 
+  async disconnectAndRemoveItemsBelongingToResource(
+    resourceUri: uri.ResourceUri
+  ): Promise<void> {
+    const connections = this.getConnections().filter(s => {
+      switch (s.kind) {
+        case 'connection.server':
+          return s.serverUri === resourceUri;
+        case 'connection.gateway':
+          return s.targetUri === resourceUri;
+        case 'connection.kube':
+          return s.kubeUri === resourceUri;
```

Comment:
```suggestion
          return s.kubeUri === resourceUri;
        default:
          return assertUnreachable(s)
```

\+ `import { assertUnreachable } from 'teleterm/ui/utils';`

I think it should be safe to do as connections are created from documents and unrecognized documents are simply ignored. So there will not be a situation where this functions runs into a connection restored from some kind of a disk state and crashes.